cmd/routed-eni-cni-plugin/cni.go

```diff
@@ -275,6 +278,34 @@ func add(args *skel.CmdArgs, cniTypes typeswrapper.CNITYPES, grpcClient grpcwrap
 	// dummy interface is appended to PrevResult for use during cleanup
 	result.Interfaces = append(result.Interfaces, dummyInterface)
 
+	if utils.IsStrictMode(r.NetworkPolicyMode) {
+		// Set up a connection to the network policy agent
+		npConn, err := grpcClient.Dial(npAgentAddress, grpc.WithTransportCredentials(insecure.NewCredentials()))
+		if err != nil {
+			log.Errorf("Failed to connect to network policy agent: %v", err)
+			return errors.Wrap(err, "add cmd: failed to connect to network policy agent backend server")
+		}
+		defer npConn.Close()
+
+		//Make a GRPC call for network policy agent
+		npc := rpcClient.NewNPBackendClient(npConn)
+
+		npr, err := npc.EnforceNpToPod(context.Background(),
+			&pb.EnforceNpRequest{
+				K8S_POD_NAME:      string(k8sArgs.K8S_POD_NAME),
+				K8S_POD_NAMESPACE: string(k8sArgs.K8S_POD_NAMESPACE),
+			})
+
+		// No need to cleanup IP and network, kubelet will send delete.
```

Comment:
Why would kubelet send a delete? Won't it just retry the ADD?

Reply:
We can discuss offline more, if Add fails then kubelet will send a delete.... but check this out -

A runtime should not call ADD twice (without an intervening DEL) for the same (CNI_CONTAINERID, CNI_IFNAME) tuple. This implies that a given container ID may be added to a specific network more than once only if each addition is done with a different interface name.

https://www.cni.dev/docs/spec/#add-add-container-to-network-or-apply-modifications

I have attached an example as well in the test..